cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c

```diff
@@ -495,6 +495,25 @@ img_mgmt_impl_upload_inspect(const struct img_mgmt_upload_req *req,
             return MGMT_ERR_ENOMEM;
         }
 
+
+#if defined(CONFIG_IMG_MGMT_REJECT_DIRECT_XIP_MISMATCHED_SLOT)
+        if (hdr->ih_flags & IMAGE_F_ROM_FIXED_ADDR) {
+            rc = flash_area_open(action->area_id, &fa);
+            if (rc) {
+                *errstr = img_mgmt_err_str_flash_open_failed;
+                return MGMT_ERR_EUNKNOWN;
+            }
+
+            if (fa->fa_off != hdr->ih_load_addr) {
+                *errstr = img_mgmt_err_str_image_bad_flash_addr;
+                return MGMT_ERR_EINVAL;
```

Comment:
This path leaves the flash area open.

Reply:
![image](https://user-images.githubusercontent.com/56024351/102192101-95067680-3eba-11eb-8206-9c0ff17caac6.png)
Sorry, fixed.